[2021-04-24 11:45:27,145] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [queued]>
[2021-04-24 11:45:27,539] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [queued]>
[2021-04-24 11:45:27,539] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-24 11:45:27,539] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-24 11:45:27,539] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-24 11:45:27,886] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-07-01T00:00:00+00:00
[2021-04-24 11:45:28,119] {standard_task_runner.py:52} INFO - Started process 216 to run task
[2021-04-24 11:45:28,587] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-07-01T00:00:00+00:00', '--job-id', '88', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmplcbujbpb', '--error-file', '/tmp/tmpsakcoswx']
[2021-04-24 11:45:28,632] {standard_task_runner.py:77} INFO - Job 88: Subtask Stage_songs
[2021-04-24 11:45:30,765] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [running]> on host 88dcd2a12842
[2021-04-24 11:45:32,760] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernest@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-07-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-07-01T00:00:00+00:00
[2021-04-24 11:45:32,821] {stage_redshift.py:38} INFO - Start implementation of StageToRedshiftOperator
[2021-04-24 11:45:33,118] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 39, in execute
    aws_hook = AwsHook(self.aws_credentials_id)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/contrib/hooks/aws_hook.py", line 41, in __init__
    super().__init__(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/amazon/aws/hooks/base_aws.py", line 360, in __init__
    raise AirflowException('Either client_type or resource_type must be provided.')
airflow.exceptions.AirflowException: Either client_type or resource_type must be provided.
[2021-04-24 11:45:33,195] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200701T000000, start_date=20210424T114527, end_date=20210424T114533
[2021-04-24 11:45:34,770] {local_task_job.py:146} INFO - Task exited with return code 1
[2021-04-26 09:08:54,007] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [queued]>
[2021-04-26 09:08:54,451] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [queued]>
[2021-04-26 09:08:54,452] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 09:08:54,452] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-26 09:08:54,452] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 09:08:54,633] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-07-01T00:00:00+00:00
[2021-04-26 09:08:54,701] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-07-01T00:00:00+00:00', '--job-id', '41', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmp9qc3wpy8', '--error-file', '/tmp/tmpb0j8q6fg']
[2021-04-26 09:08:54,707] {standard_task_runner.py:77} INFO - Job 41: Subtask Stage_songs
[2021-04-26 09:08:54,672] {standard_task_runner.py:52} INFO - Started process 122 to run task
[2021-04-26 09:08:55,530] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [running]> on host b5c40a6f08c7
[2021-04-26 09:08:56,744] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernestoffor@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-07-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-07-01T00:00:00+00:00
[2021-04-26 09:08:56,745] {stage_redshift.py:63} INFO - Start implementation of StageToRedshiftOperator
[2021-04-26 09:08:56,745] {stage_redshift.py:68} INFO - Preparing destination Redshift table
[2021-04-26 09:08:57,320] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 09:08:59,040] {dbapi.py:180} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2021-04-26 09:08:59,471] {dbapi.py:186} INFO - Rows affected: 0
[2021-04-26 09:08:59,769] {stage_redshift.py:71} INFO - Getting data from S3 to Redshift
[2021-04-26 09:08:59,897] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 09:09:01,370] {dbapi.py:180} INFO - Running statement: 
             COPY staging_songs 
             FROM 's3://udacity-dend/song_data/*/*/*/'
             FORMAT AS JSON 'JSON 'auto''
             CREDENTIALS
             'aws_access_key_id=None;aws_secret_access_key=None'
                , parameters: None
[2021-04-26 09:09:01,763] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 81, in execute
    redshift.run(execution_sql)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/hooks/dbapi.py", line 184, in run
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near "auto"
LINE 4:              FORMAT AS JSON 'JSON 'auto''
                                           ^

[2021-04-26 09:09:01,768] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200701T000000, start_date=20210426T090854, end_date=20210426T090901
[2021-04-26 09:09:01,949] {local_task_job.py:146} INFO - Task exited with return code 1
[2021-04-26 12:35:41,508] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [queued]>
[2021-04-26 12:35:42,793] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [queued]>
[2021-04-26 12:35:42,794] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 12:35:42,794] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-26 12:35:42,804] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 12:35:43,231] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-07-01T00:00:00+00:00
[2021-04-26 12:35:43,398] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-07-01T00:00:00+00:00', '--job-id', '16', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmpps17yk0d', '--error-file', '/tmp/tmpivw4hgrm']
[2021-04-26 12:35:43,323] {standard_task_runner.py:52} INFO - Started process 90 to run task
[2021-04-26 12:35:43,403] {standard_task_runner.py:77} INFO - Job 16: Subtask Stage_songs
[2021-04-26 12:35:46,548] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [running]> on host 0f1e1da1f4f6
[2021-04-26 12:35:48,217] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernestoffor@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-07-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-07-01T00:00:00+00:00
[2021-04-26 12:35:48,230] {stage_redshift.py:62} INFO - Start implementation of StageToRedshiftOperator
[2021-04-26 12:35:48,230] {stage_redshift.py:67} INFO - Preparing destination Redshift table
[2021-04-26 12:35:48,419] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 12:35:49,898] {dbapi.py:180} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2021-04-26 12:35:50,337] {dbapi.py:186} INFO - Rows affected: 0
[2021-04-26 12:35:50,640] {stage_redshift.py:70} INFO - Getting data from S3 to Redshift
[2021-04-26 12:35:50,707] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 12:35:52,151] {dbapi.py:180} INFO - Running statement: 
             COPY staging_songs 
             FROM '/offor_data/song_data/*/*/*/'
             FORMAT AS JSON 'JSON 'auto''
                  , parameters: None
[2021-04-26 12:35:52,561] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 80, in execute
    redshift.run(execution_sql)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/hooks/dbapi.py", line 184, in run
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near "auto"
LINE 4:              FORMAT AS JSON 'JSON 'auto''
                                           ^

[2021-04-26 12:35:52,590] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200701T000000, start_date=20210426T123541, end_date=20210426T123552
[2021-04-26 12:35:53,187] {local_task_job.py:146} INFO - Task exited with return code 1
[2021-04-27 20:39:20,645] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [queued]>
[2021-04-27 20:39:21,836] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [queued]>
[2021-04-27 20:39:21,837] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-27 20:39:21,837] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-27 20:39:21,837] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-27 20:39:22,808] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-07-01T00:00:00+00:00
[2021-04-27 20:39:23,289] {standard_task_runner.py:52} INFO - Started process 196 to run task
[2021-04-27 20:39:23,448] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-07-01T00:00:00+00:00', '--job-id', '75', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmpo0m9b4ik', '--error-file', '/tmp/tmp5tfr0q_t']
[2021-04-27 20:39:23,499] {standard_task_runner.py:77} INFO - Job 75: Subtask Stage_songs
[2021-04-27 20:39:26,070] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-07-01T00:00:00+00:00 [running]> on host 478c7ec349b0
[2021-04-27 20:39:30,155] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernestoffor@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-07-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-07-01T00:00:00+00:00
[2021-04-27 20:39:30,179] {stage_redshift.py:67} INFO - Start implementation of StageToRedshiftOperator
[2021-04-27 20:39:30,189] {stage_redshift.py:74} INFO - Preparing destination Redshift table
[2021-04-27 20:39:30,784] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-27 20:39:32,599] {dbapi.py:180} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2021-04-27 20:39:33,080] {dbapi.py:186} INFO - Rows affected: 0
[2021-04-27 20:39:33,412] {stage_redshift.py:77} INFO - Getting data from S3 to Redshift
[2021-04-27 20:39:37,384] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 83, in execute
    s3.get_credentials().access_key,
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/amazon/aws/hooks/base_aws.py", line 478, in get_credentials
    return session.get_credentials().get_frozen_credentials()
AttributeError: 'NoneType' object has no attribute 'get_frozen_credentials'
[2021-04-27 20:39:37,398] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200701T000000, start_date=20210427T203920, end_date=20210427T203937
[2021-04-27 20:39:38,734] {local_task_job.py:146} INFO - Task exited with return code 1
